subcerts.go

```diff
@@ -141,67 +148,71 @@ func (cred *credential) marshal() ([]byte, error) {
 	}
 
 	// Write the length of the public_key field.
-	binary.BigEndian.PutUint16(serialized[4:], uint16(len(serializedPublicKey)))
+	binary.BigEndian.PutUint16(serialized[paramsLen:], uint16(len(serializedPublicKey)))
 
-	// Write the public key.
-	return append(serialized, serializedPublicKey...), nil
+	// Write the public_key field.
+	serialized = append(serialized, serializedPublicKey...)
+	cred.raw = serialized
+	return serialized, nil
 }
 
 // unmarshalCredential decodes a credential and returns it.
 func unmarshalCredential(serialized []byte) (*credential, error) {
-	// Bytes 0-3 are the validity time field; bytes 4-6 are the length of the
-	// serialized SubjectPublicKeyInfo.
-	if len(serialized) < 6 {
+	paramsLen := 8
+
+	// Bytes 0:4 are the valid_time field, 4:6 are the scheme field, 6:8 are the
+	// version field, and 8:10 are the length of the serialized
+	// SubjectPublicKeyInfo.
```

Comment:
Bytes 10:... contain serialized SubjectPublicKeyInfo

Reply:
Removed this comment and clarified the message structure in `(cred *credential) marshal()`.